File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/ToolWindowContentPresenter.java
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow;

import com.intellij.ui.content.ContentFactory;
import com.intellij.ui.content.ContentManager;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

import static com.google.common.base.Preconditions.checkNotNull;

public class ToolWindowContentPresenter implements ToolWindowContent.Presenter {
    @Nullable
    private ToolWindowContent.View view;
    @Nonnull
    private final ContentFactory contentFactory;
    @Nonnull
    private final ContentManager contentManager;

    public ToolWindowContentPresenter(
            @Nonnull ContentFactory contentFactory,
            @Nonnull ContentManager contentManager) {
        this.contentFactory = checkNotNull(contentFactory);
        this.contentManager = checkNotNull(contentManager);
    }

    @Override
    public void setView(@Nonnull ToolWindowContent.View view) {
        this.view = checkNotNull(view);
    }

    @Override
    public void displayContent() {
        if (view != null) {
            contentManager.addContent(contentFactory.createContent(view.getContent(), "", false));
        }
    }
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/ui/toolwindow/ToolWindowContentPresenter.kt
package com.duxtinto.intellij.plugin.github.codereviews.ui.toolwindow

import com.intellij.ui.content.ContentFactory
import com.intellij.ui.content.ContentManager
import javax.inject.Inject
import javax.inject.Named

class ToolWindowContentPresenter
    @Inject
    constructor(
            private val contentFactory: ContentFactory,
            @Named("GH_Reviews") private val contentManager: ContentManager)
    : ToolWindowContent.Presenter {
    private lateinit var revieweeView: ToolWindowContent.RevieweeView
    private lateinit var reviewerView: ToolWindowContent.ReviewerView

    override fun setRevieweeView(view: ToolWindowContent.RevieweeView) {
        revieweeView = view
        contentManager.addContent(contentFactory.createContent(view.content, "Reviewee", false))
    }

    override fun setReviewerView(view: ToolWindowContent.ReviewerView) {
        reviewerView = view
        contentManager.addContent(contentFactory.createContent(reviewerView.content, "Reviewer", false))
    }
}
